### PR TITLE
Localization Troubleshooting

### PR DESCRIPTION
**Pull request by hishamco**
This addresses issue #3166, @Rick-Anderson this is WIP PR, I will keep updating this, but I have one question, how can I add header for this topic to sidebar menu?

@Rick-Anderson 	edit: [Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/troubleshoot-aspnet-core-localization?view=aspnetcore-1.0&branch=pr-en-us-10597)

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
All the `##` become `<H2>` and are automatically added to the sidebar. So `## Localization configuration issues` and the others are added.

**Comment by hishamco** [comment]
What about adding the **Troubleshoot** into ToC?

**Comment by hishamco** [comment]
What about adding the **Troubleshoot** into ToC?

**Comment by Rick-Anderson** [comment]
@hishamco 	let me know when you complete `## Missing resources` and the H2's that follow.

**Comment by hishamco** [comment]
Hope to find time today or tomorrow to update this

**Comment by Rick-Anderson** [comment]
@hishamco 	no hurry, just checking. Thanks for your help.

File: aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+**Supported Cultures in RequestCultureProvider don't match with registered once**  
+
+## Resource file naming issues
+
+Resource file naming issues are one of the common problems that we had ever seen in the community, especially the new developers. ASP.NET Core became with set of predefine rules and guidelines for localization resources file naming, which descriped in deatils in [Resource file naming](xref:fundamentals/localization?view=aspnetcore-2.2#resource-file-naming)
+
+## Missing resources
+
+Missing localization resources is one of common cases that may occur in many web applications, because the localization resource names are incorrect, the localization resources are not exist in the actual `.resx` files or resource file naming issues as its mentioned above.
```
Comment (ryanbrandenburg) [inline aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md]:
```suggestion
Common causes of resources not being found include:
```

File: aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+To avoid such issues be sure that:
```
Comment (ryanbrandenburg) [inline aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md]:
```suggestion
```

File: aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+- Your localization resource names are set correctly
```
Comment (ryanbrandenburg) [inline aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md]:
```suggestion
- Resource names are misspelled in either the resx file or the localizer request.
```

File: aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+- Your localization resources are exist in the `resx` file of the current culture of your application
```
Comment (ryanbrandenburg) [inline aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md]:
```suggestion
- The resource is missing from the resx for some languages, but exists in others.
```

File: aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+- Check the localization log messages - which they are in `Debug` level - for more details about the missing resources.
```
Comment (ryanbrandenburg) [inline aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md]:
```suggestion
If you're still having trouble check the localization log messages (which are at `Debug` log level) for more details about the missing resources.
```

File: aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+_**Hint:** In the case of `CookieRequestCultureProvider` be sure there's no single quotes surrounds the cultures inside the localization cookie value, so `c='en-UK'|uic='en-US'` is invalid cookie value, while `c=en-UK|uic=en-US` is a valid one._
```
Comment (ryanbrandenburg) [inline aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md]:
```suggestion
_**Hint:** In the case of `CookieRequestCultureProvider` be sure there's no single quotes around the cultures inside the localization cookie value. So `c='en-UK'|uic='en-US'` is an invalid cookie value, while `c=en-UK|uic=en-US` is a valid one._
```

File: aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+
+## Resources & Class Libraries issues
+
+## CustomRequestCultureProvide doesn't work as expected
```
Comment (ryanbrandenburg) [inline aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md]:
```suggestion
## CustomRequestCultureProvider doesn't work as expected
```

**Comment by ryanbrandenburg** [comment]
It's worth considering organizing this doc more as:

## Symptom 
- Possible solution 1
- Possible solution 2

currently it seems to be more along the lines of:

## Possible cause
- Symptoms and solution

Since the user is unlikely to know the cause of the problem but should know the symptoms, I think that way is more useful to them.

File: aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+{
+    services.AddLocalization(options => options.ResourcesPath = "Resources");
+
+    services.AddMvc();
+}
+```
+
+**Localization resources path not found**
+
+**Supported Cultures in RequestCultureProvider don't match with registered once**
```
[the remark this answers is not in the document]
Comment (hishamco) [reply aspnetcore/fundamentals/troubleshoot-aspnet-core-localization.md]: Nope, few months back in SimplCommerce where I found the list of supported cultures that declared in the middleware isn't match the supported cultures that given by the request localization feature

**Comment by hishamco** [comment]
Thanks for your feedback @ryanbrandenburg

**Comment by hishamco** [comment]
@ryanbrandenburg @Rick-Anderson your final review you please, if there's any missing point I should add, shoot it to me

**Comment by Rick-Anderson** [comment]
@hishamco can you batch accept all the suggestions, then I'll grab all your commits and create a branch I'll directly edit.

**Comment by hishamco** [comment]
Do you mean squash the commits into a single one?

**Comment by hishamco** [comment]
@Rick-Anderson I think you edit directly if the Allow Maintainer to Edit option is enabled

**Comment by Rick-Anderson** [comment]
> Do you mean squash the commits into a single one?

Nevermind, I was reading this wrong and thought you hadn't taken @ryanbrandenburg  suggestions.

**Comment by hishamco** [comment]
I see .. no problem, waiting for final review ...

**Comment by Rick-Anderson** [comment]
@hishamco check [this commit](https://github.com/aspnet/Docs/pull/10597/commits/99b5a0189ea7715a69ccf25c6b818e53cf01271e) to see most of my change.

One small fix in my last commit.

**Comment by hishamco** [comment]
@Rick-Anderson I didn't see this as a separated topic in the docs, seem you forgot this, BTW I already asked you for this at the beginning

**Comment by Rick-Anderson** [comment]
@hishamco  see #11669

**Pull request merged**